pax-url-aether/src/main/java/org/ops4j/pax/url/mvn/internal/config/MavenConfigurationImpl.java

```diff
@@ -582,7 +583,11 @@ private void parseProxiesFromProperty(String proxySettings, Map<String, Map<Stri
 
     private String getSettingsPath() {
```

Comment:
I am sorry I did not make myslef clear I guess, but you code will have an NPE if getSettingsFileUrl(); returns null.
I would suggest this.
```
     private String getSettingsPath() {
         URL url = getSettingsFileUrl();
        try {
            return url == null ? null : new File(url.toURI().getSchemeSpecificPart()).getAbsolutePath();
        } catch (URISyntaxException e) {//this should never happend but we handle it anyways.
            return url.getPath();
        }
     }
```